hub/api/dataset.py

```diff
@@ -97,6 +106,15 @@ def __setitem__(self, item: Union[slice, str], value):
         else:
             raise InvalidKeyTypeError(item)
 
+    __getattr__ = __getitem__
+
+    def __setattr__(self, name: str, value):
+        """Set the named attribute on the dataset"""
+        if name in DATASET_ATTRIBUTES:
```

Comment:
this will be very hard for vscode to understand i think, also arises some issues when we have meta involved. 

i think we should have all tensors be accessed as `ds["tensor"]` and all meta values be accessed as `ds["tensor"].names`, but with the `.tensor` accessor this becomes super ambiguous

Reply:
Yes, it does. Like I said, it's super hacky. Can you think of a better way to implement it? I think the fundamental ambiguity is that we need to tell between:
```
ds.mode = 'r'
ds.image = arr
```
So we can either:
- Have a reserved list of attributes (that's what this PR does)
- Check the type of the value. This approach would look a bit nicer, but forbids us from having arr-like members on Dataset and we'd still have to overload `__setattr__`